exporters/trace/src/test/java/com/google/cloud/opentelemetry/trace/EndToEndTest.java

```diff
@@ -35,35 +40,45 @@
   private static final String SPAN_NAME = "MySpanName";
   private static final long START_EPOCH_NANOS = TimeUnit.SECONDS.toNanos(3000) + 200;
   private static final long END_EPOCH_NANOS = TimeUnit.SECONDS.toNanos(3001) + 255;
-  private static final Status SPAN_DATA_STATUS = Status.ok();
+  private static final StatusData SPAN_DATA_STATUS = StatusData.ok();
   private static final String LOCALHOST = "127.0.0.1";
 
   private MockCloudTraceClient mockCloudTraceClient;
   private TraceExporter exporter;
   private Process mockServerProcess;
 
   @Before
-  public void setup() throws IOException {
-    // Find a free port to spin up our server at.
-    ServerSocket socket = new ServerSocket(0);
-    int port = socket.getLocalPort();
-    String address = String.format("%s:%d", LOCALHOST, port);
-    socket.close();
-
-    // Start the mock server. This assumes the binary is present and in $PATH.
-    // Typically, the CI will be the one that curls the binary and adds it to $PATH.
-    String[] cmdArray = new String[] {System.getProperty("mock.server.path"), "-address", address};
-    ProcessBuilder pb = new ProcessBuilder(cmdArray);
-    pb.redirectErrorStream(true);
-    mockServerProcess = pb.start();
-
-    // Setup the mock trace client.
-    mockCloudTraceClient = new MockCloudTraceClient(LOCALHOST, port);
-
-    // Block until the mock server starts (it will output the address after starting).
-    BufferedReader br =
-        new BufferedReader(new InputStreamReader(mockServerProcess.getInputStream()));
-    br.readLine();
+  public void setup() throws MockServerStartupException {
+    try {
+      // Find a free port to spin up our server at.
+      ServerSocket socket = new ServerSocket(0);
+      int port = socket.getLocalPort();
+      String address = String.format("%s:%d", LOCALHOST, port);
+      socket.close();
+
+      // Start the mock server. This assumes the binary is present and in $PATH.
+      // Typically, the CI will be the one that curls the binary and adds it to $PATH.
```

Comment:
Same as above

Reply:
same comment as above, also I'd like to share this in a test-util library component that's not released.